python/ray/experimental/channel/nccl_group_interface.py

```diff
@@ -0,0 +1,76 @@
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, Callable, Optional, Tuple
+
+import ray
+from ray.util.annotations import DeveloperAPI
+
+if TYPE_CHECKING:
+    import torch
+
+
+# Signature for a torch.Tensor allocator is:
+# (shape: Tuple[int], dtype: torch.dtype) -> torch.Tensor.
+TorchTensorAllocator = Callable[[Tuple[int], "torch.dtype"], "torch.Tensor"]
+
+
+@DeveloperAPI
+class NcclGroupInterface(ABC):
+    """
+    Interface for an actor's NCCL communicator.
+    """
+
+    @abstractmethod
+    def get_rank(self, actor: ray.actor.ActorHandle) -> int:
```

Comment:
how does it find the next rank correctly? (is there any assumption in the rank order that could be different from communicator that's passed?)

Reply:
Why do we need to get the next rank? As long as there is a map from Actor to rank it will be sufficient?